Update java allocation profiling enablement (#20460)
DataDog/documentation: content/en/profiler/enabling/java.md

The wallclock engine does not depend on the `/proc/sys/kernel/perf_event_paranoid` setting.

### Profiler allocation engine

> {{< tabs >}}

Reviewer: Suggest switching the order to be consistent with the rest of the tabs on the page. Datadog Profiler tab before JFR tab.

Author: The reason I switched the order is that the for allocation profiling we are defaulting to 'JFR' and the user needs to opt in to actually enable the 'Datadog' profiler.
If the order does not make sense, let me know and I will change it - however, it would be nice to make it very explicit that unlike eg. the Datadog CPU profiler, the Datadog allocation profiler is an opt-in.

{{% tab "JFR" %}}
The JFR based allocation profiling engine is enabled by default since JDK 16.

> The reason for not enabling it by default for JDK 8 and 11 is that an allocation intensive

Reviewer: If this suggestion alter's the message, leave it as is.
```suggestion
The reason it's not enabled by default for JDK 8 and 11 is that an allocation intensive
```

> The reason fora not enabling it by default for JDK 8 and 11 is that an allocation intensive

Reviewer 2: ```suggestion
The reason for not enabling it by default for JDK 8 and 11 is that an allocation intensive
```

> application can create a very large number of the related JFR events.

Reviewer 2: ```suggestion
application can lead to high overhead and large recording sizes.
```

> If you still want to enable it, you can do it with:

Reviewer: ```suggestion
To enable it for JDK 8 and 11, add the following:
```

```
export DD_PROFILING_ENABLED_EVENTS=jdk.ObjectAllocationInNewTLAB,jdk.ObjectAllocationOutsideTLAB
```

or:

```
-Ddd.profiling.enabled.events=jdk.ObjectAllocationInNewTLAB,jdk.ObjectAllocationOutsideTLAB
```
{{% /tab %}}

{{% tab "Datadog Profiler" %}}
_Requires JDK 11+._

The Datadog allocation profiling engine contextualizes allocation profiles, which supports allocation profiles filtered by endpoint.

> In dd-java-agent earlier than v1.17.0 it is **disabled** by default, enable it with:

Reviewer 3: ```suggestion
In dd-java-agent earlier than v1.17.0 it is **disabled** by default. Enable it with:
```

> In dd-java-agent earlier than v1.17.0 it is **disabled** by default, but you can enable it with:

Reviewer: ```suggestion
In dd-java-agent earlier than v1.17.0 it is **disabled** by default, enable it with:
```
